src/app/wallets/get-transactions-for-wallet.ts

```diff
@@ -44,20 +42,19 @@ export const getTransactionsForWallet = async (
     return PartialResult.partial(confirmedHistory.transactions, onChainTxs)
   }
 
+  const addresses = wallet.onChainAddresses()
   const filter = TxFilter({
     confirmationsLessThan: ONCHAIN_MIN_CONFIRMATIONS,
-    addresses: wallet.onChainAddresses,
+    addresses,
   })
   const pendingTxs = filter.apply(onChainTxs)
 
-  // TODO should be a service - not a function call
-  let price = await getCurrentPrice()
-  if (typeof price !== "number") {
+  let price = await PriceService().getCurrentPrice()
+  if (price instanceof Error) {
     price = NaN
```

Comment:
Minor nit for the type-checker
```suggestion
  if (price instanceof Error) {
    price = NaN as UsdPerSat
```